From: Dan Carpenter <dan.carpenter@oracle.com>
Subject: mm/mmap: delete dead code in do_brk_flags()

This code can't be reached.  It's harmless but it leads to a Smatch
warning:

    mm/mmap.c:3040 do_brk_flags() warn: ignoring unreachable code.

Link: https://lkml.kernel.org/r/YmqHrpo/cBBq6lHf@kili
Fixes: d2bbe46ab3b9 ("mm/mmap: change do_brk_flags() to expand existing VMA and add do_brk_munmap()")
Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Reviewed-by: David Hildenbrand <david@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/mmap.c |    1 -
 1 file changed, 1 deletion(-)

--- a/mm/mmap.c~mm-mmap-delete-dead-code-in-do_brk_flags
+++ a/mm/mmap.c
@@ -3031,7 +3031,6 @@ out:
 	validate_mm(mm);
 	return 0;
 
-	vm_area_free(vma);
 vma_alloc_fail:
 	vm_unacct_memory(len >> PAGE_SHIFT);
 	return -ENOMEM;
_
